solidity/test/bridge/Bridge.Wallets.test.ts

```diff
@@ -0,0 +1,629 @@
+              ).to.be.revertedWith("Wallet creation conditions are not met")
+            })
+          })
+        })
+      })
+
+      context("when wallet creation is already in progress", () => {
+        const testData = [
+          {
+            testName: "when wallet creation state is AWAITING_SEED",
+            walletCreationState: ecdsaDkgState.AWAITING_SEED,
+          },
+          {
+            testName: "when wallet creation state is AWAITING_RESULT",
+            walletCreationState: ecdsaDkgState.AWAITING_RESULT,
+          },
+          {
+            testName: "when wallet creation state is CHALLENGE",
+            walletCreationState: ecdsaDkgState.CHALLENGE,
+          },
+        ]
+
+        testData.forEach((test) => {
+          context(test.testName, () => {
+            before(async () => {
+              await createSnapshot()
+
+              walletRegistry.getWalletCreationState.returns(
+                test.walletCreationState
+              )
+            })
+
+            after(async () => {
+              walletRegistry.getWalletCreationState.reset()
+
+              await restoreSnapshot()
+            })
+
+            it("should revert", async () => {
+              await expect(
+                bridge.requestNewWallet(NO_MAIN_UTXO)
+              ).to.be.revertedWith("Wallet creation already in progress")
+            })
+          })
+        })
+      })
+    })
+  })
+
+  describe("__ecdsaWalletCreatedCallback", () => {
+    context("when called by a third party", async () => {
+      it("should revert", async () => {
+        await expect(
+          bridge
+            .connect(thirdParty)
+            .__ecdsaWalletCreatedCallback(
+              ecdsaWalletTestData.walletID,
+              ecdsaWalletTestData.publicKeyX,
+              ecdsaWalletTestData.publicKeyY
+            )
+        ).to.be.revertedWith("Caller is not the ECDSA Wallet Registry")
+      })
+    })
+
+    context("when called by the ECDSA Wallet Registry", async () => {
+      context("when called with a valid ECDSA Wallet details", async () => {
+        let tx: ContractTransaction
+
+        before(async () => {
+          await createSnapshot()
+
+          tx = await bridge
+            .connect(walletRegistry.wallet)
+            .__ecdsaWalletCreatedCallback(
+              ecdsaWalletTestData.walletID,
+              ecdsaWalletTestData.publicKeyX,
+              ecdsaWalletTestData.publicKeyY
+            )
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should register ECDSA wallet reference", async () => {
+          await expect(
+            (
+              await bridge.getRegisteredWallet(
+                ecdsaWalletTestData.pubKeyHash160
+              )
+            ).ecdsaWalletID
+          ).equals(ecdsaWalletTestData.walletID)
+        })
+
+        it("should transition wallet to Live state", async () => {
+          await expect(
```

Comment:
This `await` is probably not needed.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/172/commits/8f091ed3ca6c2ed76f61207291b5af75a05de3f2